dashboard/client/src/components/TaskTable.tsx

```diff
@@ -0,0 +1,201 @@
+    { label: "Type" },
+    { label: "Required Resources" },
+  ];
+
+  return (
+    <div>
+      <div style={{ flex: 1, display: "flex", alignItems: "center" }}>
+        <Autocomplete
+          style={{ margin: 8, width: 120 }}
+          options={Array.from(new Set(tasks.map((e) => e.task_id)))}
+          onInputChange={(_: any, value: string) => {
+            changeFilter("task_id", value.trim());
+          }}
+          renderInput={(params: TextFieldProps) => (
+            <TextField {...params} label="Task ID" />
+          )}
+        />
+        <Autocomplete
+          style={{ margin: 8, width: 120 }}
+          options={Array.from(new Set(tasks.map((e) => e.scheduling_state)))}
+          onInputChange={(_: any, value: string) => {
+            changeFilter("scheduling_state", value.trim());
+          }}
+          renderInput={(params: TextFieldProps) => (
+            <TextField {...params} label="Scheduling State" />
+          )}
+        />
+        <Autocomplete
+          style={{ margin: 8, width: 150 }}
+          defaultValue={jobId}
+          options={Array.from(new Set(tasks.map((e) => e.job_id)))}
+          onInputChange={(_: any, value: string) => {
+            changeFilter("job_id", value.trim());
+          }}
+          renderInput={(params: TextFieldProps) => (
+            <TextField {...params} label="Job Id" />
+          )}
+        />
+        <Autocomplete
+          style={{ margin: 8, width: 150 }}
+          options={Array.from(new Set(tasks.map((e) => e.name)))}
+          onInputChange={(_: any, value: string) => {
+            changeFilter("name", value.trim());
+          }}
+          renderInput={(params: TextFieldProps) => (
+            <TextField {...params} label="Name" />
+          )}
+        />
+        <Autocomplete
+          style={{ margin: 8, width: 150 }}
+          options={Array.from(new Set(tasks.map((e) => e.func_or_class_name)))}
+          onInputChange={(_: any, value: string) => {
+            changeFilter("func_or_class_name", value.trim());
+          }}
+          renderInput={(params: TextFieldProps) => (
+            <TextField {...params} label="Function or Class Name" />
+          )}
+        />
+        <TextField
+          style={{ margin: 8, width: 120 }}
+          label="Page Size"
+          size="small"
+          defaultValue={10}
+          InputProps={{
+            onChange: ({ target: { value } }) => {
+              setPageSize(Math.min(Number(value), 500) || 10);
+            },
+            endAdornment: (
+              <InputAdornment position="end">Per Page</InputAdornment>
+            ),
+          }}
+        />
+      </div>
+      <div style={{ display: "flex", alignItems: "center" }}>
+        <div>
+          <Pagination
+            page={pageNo}
+            onChange={(e, num) => setPageNo(num)}
+            count={Math.ceil(taskList.length / pageSize)}
+          />
+        </div>
+      </div>
+      <Table>
```

Comment:
TODO add a status chip later. Now it is not possible because of the field name (status chip only works with a name "state", but task APi provides "scheduling_state")